<!doctype html>
<html lang="en">
<head>
<title>CDN-scRNAseq-online</title>
<meta name="generator" content="Etherpad">
<meta name="author" content="Etherpad">
<meta name="changedby" content="Etherpad">
<meta charset="utf-8">
<style>
ol {
  counter-reset: item;
}

ol > li {
  counter-increment: item;
}

ol ol > li {
  display: block;
}

ol > li {
  display: block;
}

ol > li:before {
  content: counters(item, ".") ". ";
}

ol ol > li:before {
  content: counters(item, ".") ". ";
  margin-left: -20px;
}

</style>
</head>
<body>
Welcome to CDN mini-workshop EtherPad!!<br><br>This pad is synchronized as you type, so that everyone viewing this page sees the same text. This allows you to collaborate seamlessly on documents.<br><br>---------------------------------------------<br>Lesson: scRNA-seq analysis using Seurat&nbsp;<br>Date: 29th June 2021<br>Time: 10am to 1pm<br>---------------------------------------------<br><br>## Lesson preparation ##<br>Before the start of this lesson, please ensure that you have the required software installed on your computer.<br>Please visit <a href="https&#x3a;&#x2F;&#x2F;fursham&#x2d;h&#x2e;github&#x2e;io&#x2F;R&#x2d;datascience&#x2F;articles&#x2F;4&#x5f;scRNAseq&#x5f;seurat&#x2F;Overview&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;fursham-h.github.io&#x2F;R-datascience&#x2F;articles&#x2F;4_scRNAseq_seurat&#x2F;Overview.html</a> for details on software installation and for more information on the schedule of this class.<br><br>Looking forward to our workshop!<br><br>#########################################<br>Check-in exercise!<br>Please type the following in the space below:<br>Name; Have you used Seurat before?; Are you okay with us recording this workshop?<br><br>Fursham; Yes; Yes<br>Alexis; Yes; Yes<br>Katie; No;Yes<br>Tek; No; Yes<br>Youran: yes, yes<br>Fazal:No:yes<br>Karen; no;yes<br><br><br><br>#########################################<br><br><br>## what will we be doing today&nbsp;<br>### 1. Process scRNA-seq data using Seurat<br>### 2. Perform QC<br>### 3. Regress cell-cycle genes<br>### 4. Normalize and scale count matrix<br>### 5. Reduce dimensionality using PCA and UMAP<br>### 6. Identify clusters<br>### 7. Perform differential expression<br><br>## Overview of seurat pipeline&nbsp;<br><a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;077e0f6da54d13153aa2c88ae615a1efaa913294&#x2F;vignettes&#x2F;4&#x5f;scRNAseq&#x5f;seurat&#x2F;cropped&#x5f;seurat&#x5f;workflow&#x2e;png" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;077e0f6da54d13153aa2c88ae615a1efaa913294&#x2F;vignettes&#x2F;4_scRNAseq_seurat&#x2F;cropped_seurat_workflow.png</a><br><br>## Inputs for scRNA-seq analysis<br>### 1. A count matrix (most basic)<br>### 2. Output files from cellranger software (10x genomics)<br><br>## Data used in this lesson:<br>### Childhood Cerebellar Tumors Mirror Conserved Fetal Transcriptional Programs<br>### <a href="https&#x3a;&#x2F;&#x2F;www&#x2e;nature&#x2e;com&#x2F;articles&#x2F;s41586&#x2d;019&#x2d;1158&#x2d;7" rel="noreferrer noopener">https:&#x2F;&#x2F;www.nature.com&#x2F;articles&#x2F;s41586-019-1158-7</a><br>### Find information on downloading datasets from &quot;Data availability&quot; section<br>### <a href="https&#x3a;&#x2F;&#x2F;www&#x2e;ncbi&#x2e;nlm&#x2e;nih&#x2e;gov&#x2F;geo&#x2F;query&#x2F;acc&#x2e;cgi&#x3f;acc&#x3d;GSE118068" rel="noreferrer noopener">https:&#x2F;&#x2F;www.ncbi.nlm.nih.gov&#x2F;geo&#x2F;query&#x2F;acc.cgi?acc=GSE118068</a><br><br>## Workflow used<br># <a href="https&#x3a;&#x2F;&#x2F;satijalab&#x2e;org&#x2F;seurat&#x2F;articles&#x2F;pbmc3k&#x5f;tutorial&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;satijalab.org&#x2F;seurat&#x2F;articles&#x2F;pbmc3k_tutorial.html</a><br><br>## Useful Seurat commands<br># <a href="https&#x3a;&#x2F;&#x2F;satijalab&#x2e;org&#x2F;seurat&#x2F;articles&#x2F;essential&#x5f;commands&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;satijalab.org&#x2F;seurat&#x2F;articles&#x2F;essential_commands.html</a><br><br><br><br><br>#### PART 1: Creating Seurat Object<br><br>## Open a new R script in RStudio<br><br># Load R libraries<br>library(Seurat)<br>library(tidyverse)<br><br># To import 10x data<br>## The files have to be named:<br>### 1. barcodes.tsv.gz<br>### 2. features.tsv.gz<br>### 3. matrix.mtx.gz<br>### If your 10x files are not named this way, use ReadMtx()<br>e14.crblm &lt;- Read10X(data.dir = &quot;~&#x2F;GSE118068_embryo&#x2F;E14&#x2F;&quot;)<br><br>## 10x may also provide h5 file format<br>## use Read10X_h5() to import this<br><br># To import countmatrix<br># data &lt;- ReadMtx(mtx = &quot;path&#x2F;to&#x2F;countmatrix&quot;)<br><br># To create a Seurat Object<br>e14.seurat.crblm &lt;- CreateSeuratObject(counts = e14.crblm, project = &quot;e14&quot;)<br><br>#### PART 2: Inspecting Seurat Objects<br><br>## typing out variable name gives a description of the Seurat Object<br>e14.seurat.crblm<br><br># get cell barcodes (ID)<br>colnames(e14.seurat.crblm) %&gt;% head()<br><br># get features (genes)<br>## IMPORTANT. Try to have gene_names as features, rather than gene_id<br>rownames(e14.seurat.crblm) %&gt;% head()<br><br># View default assay&nbsp;<br>DefaultAssay(e14.seurat.crblm)<br><br># View count data<br>e14.seurat.crblm[[&quot;RNA&quot;]]@counts %&gt;% head()<br><br># View dataset metadata<br>e14.seurat.crblm[[]] %&gt;% head()<br><br>## or<br>e14.seurat.crblm@meta.data %&gt;% head()<br><br># To add new metadata column<br>e14.seurat.crblm$region &lt;- &quot;Cerebellum&quot;<br>## same as&nbsp;<br>## e14.seurat.crblm[[&quot;region&quot;]] &lt;- &quot;Cerebellum&quot;<br><br><br>#### Activity 1 ####<br><br># import E16 and E18 datasets into R and create Seurat Objects for each&nbsp;<br># datasets (name them e16.seurat.crblm &amp; e18.seurat.crblm)<br><br><br># Ans:<br>e16.crblm &lt;- Read10X(data.dir = &quot;~&#x2F;GSE118068_embryo&#x2F;E16&#x2F;&quot;)<br>e16.seurat.crblm &lt;- CreateSeuratObject(counts = e16.crblm, project = &quot;e16&quot;)<br><br>e18.crblm &lt;- Read10X(data.dir = &quot;~&#x2F;GSE118068_embryo&#x2F;E18&#x2F;&quot;)<br>e18.seurat.crblm &lt;- CreateSeuratObject(counts = e18.crblm, project = &quot;e18&quot;)<br><br>###################<br><br><br># PART 3: Combining Seurat Objects<br><br><br># simple merging of 3 objects<br>seurat.merged &lt;- merge(e14.seurat.crblm, y = c(e16.seurat.crblm, e18.seurat.crblm), add.cell.ids = c(&quot;e14&quot;, &quot;e16&quot;, &quot;e18&quot;))&nbsp;<br><br><br>colnames(seurat.merged) %&gt;% head()<br>colnames(seurat.merged) %&gt;% tail()<br><br>#### Activity 2 ####<br><br># type an &quot;x&quot; beside your answer<br># What is the total number of cells?<br><br>## a) 27998<br>## b) 6068<br>## c) 19401 xxxx<br>## d) 7383<br><br># Ans: c<br><br># check by typing:<br>seurat.merged<br>######&gt;&gt;&gt;&gt;&nbsp;<br><br><br># What is the highest number of genes detected in a cell?<br><br>## a) 7168 x<br>## b) 10288<br>## c) 3986<br>## d) 4508<br><br># Ans: a<br><br># check by typing:<br>seurat.merged$nFeature_RNA %&gt;% max()<br><br>## PRO TIP:<br>To identify which cell has the highest number of genes, type:<br>seurat.merged[[]][which(seurat.merged$nFeature_RNA == max(seurat.merged$nFeature_RNA)),]<br>######&gt;&gt;&gt;&gt;&nbsp;<br><br>###################<br><br># check metadata<br>seurat.merged[[]] %&gt;% head()<br>#or<br>seurat.merged@meta.data %&gt;% head()<br><br>seurat.merged[[]] %&gt;% tail()<br>#or<br>seurat.merged@meta.data %&gt;% tail()<br><br><br># Create region metadata for all samples<br>seurat.merged$region &lt;- &quot;Cerebellum&quot;<br><br><br># check identity of cells<br>Idents(seurat.merged) %&gt;% levels()&nbsp;<br><br>###################<br><br># PART 4: QC of scRNAseq dataset<br><br>## Check if Seurat objects contain mitochondrial genes<br>str_subset(rownames(seurat.merged), &quot;^mt-&quot;)<br><br><br>## to calculate the percentage of reads that mapped to mitochondrial genome<br>seurat.merged$percent.mt &lt;- PercentageFeatureSet(seurat.merged, pattern = &quot;^mt-&quot;)<br><br><br># To create violin plot of continuous values<br>VlnPlot(seurat.merged, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;))<br><br><br>Idents(seurat.merged) &lt;- seurat.merged$region<br>VlnPlot(seurat.merged, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;))<br><br><br><br>## Plot two-way scatter graph<br>plot1 &lt;- FeatureScatter(seurat.merged, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;)<br>plot2 &lt;- FeatureScatter(seurat.merged, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;)<br>plot1 + plot2<br><br><br># Drawing lines on plot to visualize thresholds<br>## TIP: these are ggplots, so you can amend them using ggplot functions<br>plot2 + geom_hline(yintercept = 300)<br><br>plot2 + geom_hline(yintercept = 500) + geom_vline(xintercept = 21000) &lt;&lt; run thisss<br><br>plot1 + geom_hline(yintercept = 5)<br><br>## Filtering out poor quality cells&nbsp;<br>### !!This will overwrite the imported data!!<br><br><br>seurat.merged &lt;- subset(seurat.merged, subset = nFeature_RNA &gt; 500 &amp; nCount_RNA &lt; 21000 &amp; percent.mt &lt; 5)<br><br><br><br># check plot again<br>plot1 &lt;- FeatureScatter(seurat.merged, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;)<br>plot2 &lt;- FeatureScatter(seurat.merged, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;)<br>plot1 + plot2<br><br><br><br># PART 5: Normalizing data, scaling data and identifying Highly Variable Features<br>## <a href="https&#x3a;&#x2F;&#x2F;satijalab&#x2e;org&#x2F;seurat&#x2F;articles&#x2F;sctransform&#x5f;vignette&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;satijalab.org&#x2F;seurat&#x2F;articles&#x2F;sctransform_vignette.html</a><br><br><br>seurat.merged &lt;- SCTransform(seurat.merged, verbose = TRUE)<br><br><br>DefaultAssay(seurat.merged)<br><br><br>Break starts at 11:33,<br>Resume at 11 48<br><br><br><br>## PART 6: Perform Linear dimensional reduction<br><br>### PCA vs UMAP vs tSNE<br># <a href="https&#x3a;&#x2F;&#x2F;towardsdatascience&#x2e;com&#x2F;dimensionality&#x2d;reduction&#x2d;for&#x2d;data&#x2d;visualization&#x2d;pca&#x2d;vs&#x2d;tsne&#x2d;vs&#x2d;umap&#x2d;be4aa7b1cb29" rel="noreferrer noopener">https:&#x2F;&#x2F;towardsdatascience.com&#x2F;dimensionality-reduction-for-data-visualization-pca-vs-tsne-vs-umap-be4aa7b1cb29</a><br><br>#PCA<br>seurat.merged &lt;- RunPCA(seurat.merged, features = VariableFeatures(object = seurat.merged))<br><br>DimPlot(seurat.merged)<br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;)<br><br><br># OPTIONAL: Regressing cell cycle genes<br>cc.genes %&gt;% head()<br>s.genes &lt;- str_to_title(cc.genes$s.genes)<br>g2m.genes &lt;- str_to_title(cc.genes$g2m.genes)<br><br>seurat.merged &lt;- CellCycleScoring(seurat.merged, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)<br><br>seurat.merged &lt;- RunPCA(seurat.merged, features = c(s.genes, g2m.genes))<br>pca.before &lt;- DimPlot(seurat.merged, group.by = &quot;Phase&quot;)<br><br># Regressing cell cycle genes<br>seurat.merged &lt;- SCTransform(seurat.merged, vars.to.regress = c(&quot;S.Score&quot;, &quot;G2M.Score&quot;), verbose = TRUE)<br>seurat.merged &lt;- RunPCA(seurat.merged, features = c(s.genes, g2m.genes))<br>DimPlot(seurat.merged, group.by = &quot;Phase&quot;)<br><br>seurat.merged &lt;- RunPCA(seurat.merged, features = VariableFeatures(object = seurat.merged))<br>pca.after &lt;- DimPlot(seurat.merged, group.by = &quot;Phase&quot;)<br><br><br><br># OPTIONAL: Determine dataset dimensionality<br># <a href="https&#x3a;&#x2F;&#x2F;satijalab&#x2e;org&#x2F;seurat&#x2F;articles&#x2F;pbmc3k&#x5f;tutorial&#x2e;html&#x23;determine&#x2d;the&#x2d;dimensionality&#x2d;of&#x2d;the&#x2d;dataset&#x2d;1" rel="noreferrer noopener">https:&#x2F;&#x2F;satijalab.org&#x2F;seurat&#x2F;articles&#x2F;pbmc3k_tutorial.html#determine-the-dimensionality-of-the-dataset-1</a><br><br><br><br># PART 8: Perform non-linear dimensional reduction (UMAP&#x2F;TSNE)<br><br>seurat.merged &lt;- RunUMAP(seurat.merged, dims = 1:30, verbose = TRUE)<br>## seurat.merged &lt;- RunTSNE(seurat.merged, dims = 1:30)<br>DimPlot(seurat.merged)<br>#DimPlot(seurat.merged, reduction = &quot;tsne&quot;, group.by = &quot;orig.ident&quot;)<br><br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;)<br><br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T)<br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T) + NoLegend()<br><br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T, pt.size = 1) + NoLegend()<br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T, label.size = 10) + NoLegend()<br><br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T, cols = c(&quot;Black&quot;, &quot;Grey&quot;, &quot;Red&quot;)) + NoLegend()<br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;, label = T, repel = T, cols = &quot;Set2&quot;)<br><br>## ADVANCE mod<br>DimPlot(seurat.merged, group.by = &quot;orig.ident&quot;,&nbsp;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; cells.highlight = rownames(seurat.merged@meta.data[seurat.merged$orig.ident == &quot;e14&quot;,]),&nbsp;<br>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; cols.highlight = &quot;cornflowerblue&quot;, sizes.highlight = 1)<br><br><br># Plot gene expressions on UMAP<br>FeaturePlot(seurat.merged, &quot;Rora&quot;)<br>FeaturePlot(seurat.merged, &quot;Rora&quot;, split.by = &quot;orig.ident&quot;)<br><br>FeaturePlot(seurat.merged, c(&quot;Rora&quot;, &quot;Pax2&quot;, &quot;Pax3&quot;, &quot;Atoh1&quot;, &quot;Pax6&quot;, &quot;Lhx9&quot;))<br>## Same plot modifications can be done as DimPlot<br><br># Visualize gene expression as violin plot<br>VlnPlot(seurat.merged, &quot;Rora&quot;)<br>VlnPlot(seurat.merged, &quot;Rora&quot;, group.by = &quot;orig.ident&quot;)<br>VlnPlot(seurat.merged, c(&quot;Rora&quot;, &quot;Pax2&quot;, &quot;Pax3&quot;, &quot;Atoh1&quot;, &quot;Pax6&quot;, &quot;Lhx9&quot;))<br><br><br># PART 9: Cell clustering<br><br>seurat.merged &lt;- FindNeighbors(seurat.merged, dims = 1:30)<br>seurat.merged &lt;- FindClusters(seurat.merged, resolution = 0.2)<br>DimPlot(seurat.merged)<br>DimPlot(seurat.merged, label = T, repel = T) + NoLegend()<br><br>#PART 10: Identifying gene markers<br><br>cluster.markers &lt;- FindAllMarkers(seurat.merged, only.pos = TRUE, min.pct = 0.25, logfc.threshold = 0.25)<br><br>top3 &lt;- cluster.markers %&gt;%&nbsp;<br>&nbsp;&nbsp;&nbsp; group_by(cluster) %&gt;%&nbsp;<br>&nbsp;&nbsp;&nbsp; top_n(n = 3, wt = avg_log2FC)<br><br>View(top3)<br><br>DoHeatmap(seurat.merged, features = top3$gene) + NoLegend()<br>DoHeatmap(seurat.merged, subset = seurat_clusters %in% c(4,7)), features = top3$gene) + NoLegend()<br><br># PART 11: Differential gene expression<br><br>clust4_vs_clust7 &lt;- FindMarkers(seurat.merged, ident.1 = 4, ident.2 = 7, min.pct = 0.2)<br><br>VlnPlot(seurat.merged, &quot;Id2&quot;,idents = c(4,7))<br><br><br>########################################<br><br><br>## Same workflow can be used for scATAC-seq data<br>## More specific tools have been developed:<br>### 1. chromVar: <a href="https&#x3a;&#x2F;&#x2F;github&#x2e;com&#x2F;GreenleafLab&#x2F;chromVAR" rel="noreferrer noopener">https:&#x2F;&#x2F;github.com&#x2F;GreenleafLab&#x2F;chromVAR</a><br>### 2. archR: <a href="https&#x3a;&#x2F;&#x2F;www&#x2e;archrproject&#x2e;com&#x2F;" rel="noreferrer noopener">https:&#x2F;&#x2F;www.archrproject.com&#x2F;</a><br><br><br>## Other tools:<br># monocle3: <a href="https&#x3a;&#x2F;&#x2F;cole&#x2d;trapnell&#x2d;lab&#x2e;github&#x2e;io&#x2F;monocle3&#x2F;" rel="noreferrer noopener">https:&#x2F;&#x2F;cole-trapnell-lab.github.io&#x2F;monocle3&#x2F;</a><br># scanpy: <a href="https&#x3a;&#x2F;&#x2F;scanpy&#x2e;readthedocs&#x2e;io&#x2F;en&#x2F;stable&#x2F;" rel="noreferrer noopener">https:&#x2F;&#x2F;scanpy.readthedocs.io&#x2F;en&#x2F;stable&#x2F;</a><br># URD: <a href="https&#x3a;&#x2F;&#x2F;github&#x2e;com&#x2F;farrellja&#x2F;URD" rel="noreferrer noopener">https:&#x2F;&#x2F;github.com&#x2F;farrellja&#x2F;URD</a><br><br><br>### Other analyses:<br># Trajectory analyses: <a href="https&#x3a;&#x2F;&#x2F;satijalab&#x2e;org&#x2F;signac&#x2F;articles&#x2F;monocle&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;satijalab.org&#x2F;signac&#x2F;articles&#x2F;monocle.html</a><br># Probabilistic assigning of cell identity: <a href="https&#x3a;&#x2F;&#x2F;github&#x2e;com&#x2F;Irrationone&#x2F;cellassign" rel="noreferrer noopener">https:&#x2F;&#x2F;github.com&#x2F;Irrationone&#x2F;cellassign</a><br># Identifying gene programs: <a href="https&#x3a;&#x2F;&#x2F;github&#x2e;com&#x2F;dylkot&#x2F;cNMF" rel="noreferrer noopener">https:&#x2F;&#x2F;github.com&#x2F;dylkot&#x2F;cNMF</a><br># covariates.&nbsp;<br><br><br><br># Closing activity&nbsp;<br><br>#1) What is the one thing you learn today?<br><br>#2) What is the one thing this workshop can improve on?<br><br><br><br><br>
</body>
</html>
