<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Workshop 3: Intro to RNAseq analysis • CDN.R.Datascience</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Workshop 3: Intro to RNAseq analysis">
<meta property="og:description" content="CDN.R.Datascience">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">CDN.R.Datascience</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    Lessons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/1_R_basics_for_data_analysis/Overview.html">1) R Basics for Data Analysis</a>
    </li>
    <li>
      <a href="../../articles/2_Data_visualisation_using_R/Overview.html">2) Data visualisation using R</a>
    </li>
    <li>
      <a href="../../articles/3_RNAseq_workflow/Overview.html">2) Introduction to RNAseq data analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fursham-h/R-datascience">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Workshop 3: Intro to RNAseq analysis</h1>
                        <h4 class="author">Fursham Hamid</h4>
            
      
      
      <div class="hidden name"><code>Overview.Rmd</code></div>

    </div>

    
    
<blockquote>
<h3 id="overview">Overview</h3>
<p><br><strong>Duration</strong>: 4 hours (over 2 days) <br><br></p>
<p><strong>Objectives</strong></p>
<ol style="list-style-type: decimal">
<li>Describe key reasons for learning shell.</li>
<li>Navigate your file system using the command line.</li>
<li>Interconvert between absolute and relative paths.</li>
<li>View, search within, copy, move, and rename files. Create new directories.</li>
<li>Use wildcards (*) to perform operations on multiple files.</li>
<li>Employ the <code>grep</code> command to search for information within files.</li>
<li>Print the results of a command to a file.</li>
<li>Explain how a FASTQ file encodes per-bClean FASTQ reads using Trimmomatic.ase quality scores.</li>
<li>Interpret a FastQC plot summarizing per-base quality across all reads.</li>
<li>Clean FASTQ reads using Trimmomatic.</li>
<li>Align and quantify reads using Kallisto <br>
</li>
</ol>
</blockquote>
<div id="before-starting-the-lesson" class="section level2">
<h2 class="hasAnchor">
<a href="#before-starting-the-lesson" class="anchor"></a>Before starting the lesson</h2>
<p>Please make sure that you have the following software installed on your computer:</p>
<div id="the-bash-shell" class="section level3">
<h3 class="hasAnchor">
<a href="#the-bash-shell" class="anchor"></a><strong>The Bash Shell</strong>
</h3>
<p>Bash is a commonly-used shell that gives you the power to do simple tasks more quickly.</p>
</div>
<div id="section" class="section level3 tabset">
<h3></h3>
<div id="windows" class="section level4">
<h4 class="hasAnchor">
<a href="#windows" class="anchor"></a>Windows</h4>
<p><a href="https://www.youtube.com/watch?v=339AEqk9c-8">Video Tutorial</a></p>
<ol style="list-style-type: decimal">
<li>Download the Git for Windows installer.</li>
<li>Run the installer and follow the steps below:
<ol style="list-style-type: decimal">
<li>Click on “Next” four times (two times if you’ve previously installed Git). You don’t need to change anything in the Information, location, components, and start menu screens.</li>
<li>From the dropdown menu select “Use the nano editor by default” and click on “Next”.</li>
<li>Ensure that “Git from the command line and also from 3rd-party software” is selected and click on “Next”. (If you don’t do this Git Bash will not work properly, requiring you to remove the Git Bash installation, re-run the installer and to select the “Git from the command line and also from 3rd-party software” option.)</li>
<li>Ensure that “Use the native Windows Secure Channel library” is selected and click on “Next”.</li>
<li>Ensure that “Checkout Windows-style, commit Unix-style line endings” is selected and click on “Next”.</li>
<li>Ensure that “Use Windows’ default console window” is selected and click on “Next”.</li>
<li>Ensure that “Enable file system caching” and “Enable Git Credential Manager” are selected and click on “Next”.</li>
<li>Click on “Install”.</li>
<li>Click on “Finish”.</li>
</ol>
</li>
<li>If your “HOME” environment variable is not set (or you don’t know what this is):
<ol style="list-style-type: decimal">
<li>Open command prompt (Open Start Menu then type cmd and press [Enter])</li>
<li>Type the following line into the command prompt window exactly as shown: <code>setx HOME "%USERPROFILE%"</code>
</li>
<li>Press [Enter], you should see SUCCESS: Specified value was saved.</li>
<li>Quit command prompt by typing exit then pressing [Enter]</li>
</ol>
</li>
</ol>
<p>This will provide you with both Git and Bash in the Git Bash program.</p>
</div>
<div id="macos" class="section level4">
<h4 class="hasAnchor">
<a href="#macos" class="anchor"></a>MacOS</h4>
<p>The default shell in some versions of macOS is Bash, and Bash is available in all versions, so no need to install anything. You access Bash from the Terminal (found in <code>/Applications/Utilities</code>). See the Git installation video tutorial for a n example on how to open the Terminal. You may want to keep Terminal in your dock for this workshop.</p>
<p>To see if your default shell is Bash type <code>echo $SHELL</code> in Terminal and press the enter/return key. If the message printed does not end with ‘/bash’ then your default is something else and you can run Bash by typing <code>bash</code>.</p>
</div>
<div id="linux" class="section level4">
<h4 class="hasAnchor">
<a href="#linux" class="anchor"></a>Linux</h4>
<p>The default shell is usually Bash and there is usually no need to install anything.</p>
<p>To see if your default shell is Bash type <code>echo $SHELL</code> in Terminal and press the enter/return key. If the message printed does not end with ‘/bash’ then your default is something else and you can run Bash by typing <code>bash</code>.</p>
</div>
</div>
<div id="section-1" class="section level3">
<h3></h3>
</div>
<div id="software-to-process-rnaseq-data" class="section level3">
<h3 class="hasAnchor">
<a href="#software-to-process-rnaseq-data" class="anchor"></a><strong>Software to process RNAseq data</strong>
</h3>
<table class="table">
<colgroup>
<col width="7%">
<col width="23%">
<col width="23%">
<col width="23%">
<col width="23%">
</colgroup>
<thead><tr class="header">
<th align="left">Software</th>
<th align="left">Version</th>
<th align="left">Manual</th>
<th align="left">Available for</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></td>
<td align="left">0.11.9</td>
<td align="left"><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">Link</a></td>
<td align="left">Linux, MacOS, Windows</td>
<td align="left">Quality control tool for high throughput sequence data.</td>
</tr>
<tr class="even">
<td align="left"><a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a></td>
<td align="left">0.39</td>
<td align="left"><a href="http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf">Link</a></td>
<td align="left">Linux, MacOS, Windows</td>
<td align="left">A flexible read trimming tool for Illumina NGS data.</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://pachterlab.github.io/kallisto/">Kallisto</a></td>
<td align="left">0.46.1</td>
<td align="left"><a href="https://pachterlab.github.io/kallisto/manual">Link</a></td>
<td align="left">Linux, MacOS, Windows</td>
<td align="left">Quantifying abundances of transcripts from bulk and single-cell RNA-Seq data.</td>
</tr>
</tbody>
</table>
<p>The above software are needed on day 2 of the workshop.Detailed instructions on installing these software can be found <a href="https://fursham-h.github.io/R-datascience/articles/2_RNAseq_workflow/installations.html">here</a>. We will explain how to install these software on day 1 of the workshop.</p>
</div>
<div id="etherpad" class="section level3">
<h3 class="hasAnchor">
<a href="#etherpad" class="anchor"></a><strong>Etherpad</strong>
</h3>
<p>During the lesson, we will be using etherpad, an online collaborative platform for our discussions and activities. The link to this workshop’s etherpad page can be found <a href="https://etherpad.wikimedia.org/p/CDN-RNAseqIntro-online">here</a>. You may bookmark this page before the lesson commence!</p>
<p><br></p>
</div>
</div>
<div id="lesson-schedule" class="section level2">
<h2 class="hasAnchor">
<a href="#lesson-schedule" class="anchor"></a>Lesson schedule</h2>
<p><strong>Day 1</strong></p>
<table class="table">
<colgroup>
<col width="13%">
<col width="43%">
<col width="43%">
</colgroup>
<tbody>
<tr class="odd">
<td align="left">10:00</td>
<td align="left"><a href="https://datacarpentry.org/shell-genomics/01-introduction/index.html">1. Introducing the Shell</a></td>
<td align="left">What is a command shell and why would I use one? <br>How can I move around on my computer? <br> How can I see what files and directories I have? <br> How can I specify the location of a file or directory on my computer?</td>
</tr>
<tr class="even">
<td align="left">10:20</td>
<td align="left"><a href="https://datacarpentry.org/shell-genomics/02-the-filesystem/index.html">2. Navigating files and directories</a></td>
<td align="left">How can I perform operations on files outside of my working directory? <br> What are some navigational shortcuts I can use to make my work more efficient?</td>
</tr>
<tr class="odd">
<td align="left">10:40</td>
<td align="left"><a href="https://datacarpentry.org/shell-genomics/03-working-with-files/index.html">3. Working with files and directories</a></td>
<td align="left">How can I view and search file contents? <br> How can I create, copy and delete files and directories?</td>
</tr>
<tr class="even">
<td align="left">10:50</td>
<td align="left"><strong>Break</strong></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">11:00</td>
<td align="left">4. Retrieving and inspecting RNAseq files</td>
<td align="left">How do I download files from the internet? <br> How do I compress/decompress large files? <br> How do I preview large files in Shell?</td>
</tr>
<tr class="even">
<td align="left">11:30</td>
<td align="left">5. Overview of RNAseq pipeline and installing softwares</td>
<td align="left">What is the general outline of an RNAseq data analysis? <br> How do I install command-line software?</td>
</tr>
<tr class="odd">
<td align="left">12:00</td>
<td align="left">Finish</td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
