<!doctype html>
<html lang="en">
<head>
<title>CDN-RNAseqIntro2-online</title>
<meta name="generator" content="Etherpad">
<meta name="author" content="Etherpad">
<meta name="changedby" content="Etherpad">
<meta charset="utf-8">
<style>
ol {
  counter-reset: item;
}

ol > li {
  counter-increment: item;
}

ol ol > li {
  display: block;
}

ol > li {
  display: block;
}

ol > li:before {
  content: counters(item, ".") ". ";
}

ol ol > li:before {
  content: counters(item, ".") ". ";
  margin-left: -20px;
}

</style>
</head>
<body>
Welcome to CDN mini-workshop EtherPad!!<br><br>This pad is synchronized as you type, so that everyone viewing this page sees the same text. This allows you to collaborate seamlessly on documents.<br><br>---------------------------------------------<br>Lesson: Introduction to RNAseq data analysis [Part 2]<br>Date: 25th May 2021<br>Time: 10am to 12pm<br>---------------------------------------------<br><br>## Lesson preparation ##<br>Before the start of this lesson, please ensure that you have the required software installed on your computer.<br>Please visit <a href="https&#x3a;&#x2F;&#x2F;fursham&#x2d;h&#x2e;github&#x2e;io&#x2F;R&#x2d;datascience&#x2F;articles&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;Overview&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;fursham-h.github.io&#x2F;R-datascience&#x2F;articles&#x2F;3_RNAseq_workflow&#x2F;Overview.html</a> for details on software installation and for more information on the schedule of this class.<br><br>Looking forward to our workshop!<br><br>-----------------------------------------------------------------<br><strong>Cheat sheet&nbsp;</strong><br><br><strong>- list of commands we&#x27;ve learned so far</strong><br><br>`<strong>pwd</strong>` print working directory<br>`<strong>cd</strong>` change directory (e.g., `cd ~&#x2F;Desktop` to go to Desktop folder in your home directory)<br>`<strong>ls</strong>` list content of directory (e.g., `ls -l Downloads` to list content of folder [Downloads] in long format)<br>`<strong>man [command]</strong>` or `<strong>[command] --help</strong>` get help (e.g., `man ls` or `ls --help` to get help on [ls] command)<br>`<strong>mkdir</strong>` make new directory (e.g., `mkdir RNAseq_workshop` to create new folder [RNAseq_workshop])<br>`<strong>nano</strong>` open text editor (e.g., `nano test.txt` to create a new text file called [test.txt] in the current folder)<br>`<strong>mv</strong>` move or rename (e.g., `mv test.txt test_new.txt` to rename the file [test.txt] to [test_new.txt] in the current folder)<br>`<strong>cp</strong>` copy (e.g., `cp test_new.txt RNAseq_workshop&#x2F;draft_new.txt` to copy the file [test_new.txt] to a folder called [RNAseq_workshop])<br>`<strong>rm</strong>` remove (e.g., `rm test_new.txt` to remove the file [test_new.txt] from the current folder)<br>`<strong>wc</strong>` word count (e.g., `wc -l file.txt` to count the lines in the file [file.txt])<br>`<strong>cat</strong>` print to screen (Be aware that cat will print out the entire content, even if it is a big file)<br>`<strong>less</strong>` to explore the content of a file (arrows up&#x2F;down to scroll, q to exit)<br>`<strong>head</strong>` to print to screen the beginning of a file (e.g., `head -n 2 file.txt` print the first 2 lines of the file)<br>`<strong>tail</strong>` to print to screen the end of a file (e.g., `tail -n 2 file.txt` print the last 2 lines of the file)<br>`<strong>grep</strong>` to find a matching pattern in a file (e.g., `grep NN SRR7311317_1.fastq` find the N doublets in the RNA-seq file [ SRR7311317_1.fastq])<br>`<strong>&gt;</strong>` or `<strong>&gt;&gt;</strong>` to redirect the output of a command to a file (either overwrite or append) (e.g., `grep NN SRR7311317_1.fastq &gt; Ns.txt` write in a file called [SRR7311317_1.fastq] the output of the grep search)<br>`<strong>|</strong>` pipe, direct the output of a command as input to the next (e.g., `grep NN SRR7311317_1.fastq | wc -l` search the file [SRR7311317_1.fastq] for N doublets and then count the number of lines)<br><strong>`gunzip`</strong> and<strong> `gzip`</strong> to decompress and compress .gz files respectively (e.g., `gunzip SRR7311317_1.fastq.gz` to unzip the file [SRR7311317_1.fastq.gz] and `gzip SRR7311317_1.fastq` to compress the file [SRR7311317_1.fastq])<br>`curl` to download files from the internet. (e.g., `curl -O <a href="ftp&#x3a;&#x2F;&#x2F;ftp&#x2e;sra&#x2e;ebi&#x2e;ac&#x2e;uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317&#x5f;1&#x2e;fastq&#x2e;gz" rel="noreferrer noopener">ftp:&#x2F;&#x2F;ftp.sra.ebi.ac.uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317_1.fastq.gz</a>` to download the FASTQ file to the current working directory as the original name of the file [SRR7311317_1.fastq.gz])<br><br><strong>- special hotkeys&#x2F;characters we&#x27;ve learned so far</strong><br>`Ctrl+z` or `Command+z` to force quit a running command. Useful when you accidentally run `cat` on a large file.<br>`~` shortcut to the home directory<br>`<strong>*</strong>` wildcards for any number of characters or one single character (e.g., `ls -l &#x2F;bin&#x2F;b*` to list all files beginning with &#x27;b&#x27; in the &#x2F;bin&#x2F; directory)<br><br>--------------------------------------------------------------<br><br><strong>Lesson preparations</strong><br><br><ol start="1" class="number"><li>Please make sure that you have the folders and sub-folders we created last week in your home folder. As a reminder, the folder structure should look like this:<ol start="1" class="number"><li>&nbsp;~</li><li>&nbsp;&#8627; RNAseq_workshop</li><li>&nbsp;&nbsp; &#8627; logs</li><li>&nbsp;&nbsp; &#8627; metadata</li><li>&nbsp;&nbsp; &#8627; originals</li><li>&nbsp;&nbsp;&nbsp;&nbsp; &#8627; FASTQ</li><li>&nbsp;&nbsp; &#8627; outputs&nbsp;&nbsp;&nbsp;</li><li>&#8627; scripts</ol><li>Ensure that you have both the RNAseq FASTQ files [SRR7311317_1.fastq.gz and SRR7311317_2.fastq.gz] in your ~&#x2F;RNAseq_workshop&#x2F;FASTQ folder. These files have to compressed (.gz), so please run `gzip SRR7311317_1.fastq` if you left that file in the uncompressed state. For those who wish to redownload the file, you may run the following in your terminal&#x2F;git bash:</ol><br>cd ~&#x2F;RNAseq_workshop&#x2F;originals&#x2F;FASTQ<br>curl -O <a href="ftp&#x3a;&#x2F;&#x2F;ftp&#x2e;sra&#x2e;ebi&#x2e;ac&#x2e;uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317&#x5f;1&#x2e;fastq&#x2e;gz" rel="noreferrer noopener">ftp:&#x2F;&#x2F;ftp.sra.ebi.ac.uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317_1.fastq.gz</a><br>curl -O <a href="ftp&#x3a;&#x2F;&#x2F;ftp&#x2e;sra&#x2e;ebi&#x2e;ac&#x2e;uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317&#x5f;2&#x2e;fastq&#x2e;gz" rel="noreferrer noopener">ftp:&#x2F;&#x2F;ftp.sra.ebi.ac.uk&#x2F;vol1&#x2F;fastq&#x2F;SRR731&#x2F;007&#x2F;SRR7311317&#x2F;SRR7311317_2.fastq.gz</a><br><br><ol start="1" class="number"><li>We will use 3 command-line tools; FastQC, Trimmomatic and Kallisto. Please make sure that you have these software installed on your computer if you wish to follow along the workshop. Here is the link to instructions on installing these tools [<a href="https&#x3a;&#x2F;&#x2F;fursham&#x2d;h&#x2e;github&#x2e;io&#x2F;R&#x2d;datascience&#x2F;articles&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;installations&#x2e;html&#x23;bashday2" rel="noreferrer noopener">https:&#x2F;&#x2F;fursham-h.github.io&#x2F;R-datascience&#x2F;articles&#x2F;3_RNAseq_workflow&#x2F;installations.html#bashday2</a>]. <strong>Please do not hesitate to email me if you are having problems with installation!</strong></li><li>Lastly (I promise), please download this file (<a href="https&#x3a;&#x2F;&#x2F;emckclac&#x2d;my&#x2e;sharepoint&#x2e;com&#x2F;&#x3a;u&#x3a;&#x2F;g&#x2F;personal&#x2F;k1768990&#x5f;kcl&#x5f;ac&#x5f;uk&#x2F;EZ5if5x7dbFMl9LzP&#x5f;&#x2d;9BPoBd7Eh3YzIAP7MtoEn&#x5f;IzOQA&#x3f;e&#x3d;hbQdER&#x29;" rel="noreferrer noopener">https:&#x2F;&#x2F;emckclac-my.sharepoint.com&#x2F;:u:&#x2F;g&#x2F;personal&#x2F;k1768990_kcl_ac_uk&#x2F;EZ5if5x7dbFMl9LzP_-9BPoBd7Eh3YzIAP7MtoEn_IzOQA?e=hbQdER)</a> to your ~&#x2F;RNAseq_workshop&#x2F;originals&#x2F; folder. The file is rather large (2GB), so please make sure you have space on your computer to store it temporarily. I will explain the purpose of this file next week!</ol><br><br><br>#########################################<br>Check-in exercise!<br>Please type the following in the space below:<br>Name; Favourite subject; Are you okay with me recording this session?<br><br><br>Morning! Benedikt Rome Yes<br>Maria; History; Yes<br>Grant<br>Richard, N&#x2F;A, Yes<br>Alexis, Crafts, yes<br>Megan, Art, yes<br>Youran, Biologu, yes<br>Katie, Maths, Yes<br>Fazal-<br>Karen, maths, yes<br>#########################################<br><br>## Objectives for this session:<br># 1) To check the quality of our reads<br># 2) To write a simple bash script to automate commands<br># 3) To trim our reads<br># 4) To align reads to mouse transcriptome and quantify transcript expression<br><br># Lesson schedule:<br># <a href="https&#x3a;&#x2F;&#x2F;fursham&#x2d;h&#x2e;github&#x2e;io&#x2F;R&#x2d;datascience&#x2F;articles&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;Overview&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;fursham-h.github.io&#x2F;R-datascience&#x2F;articles&#x2F;3_RNAseq_workflow&#x2F;Overview.html</a><br><br>## RECAP OF COMMANDS learnt so far<br>###### DON&#x27;T have to follow along!!<br>pwd<br>cd ~&#x2F;RNAseq_workshop&#x2F;originals<br>ls -l<br>ls -l FASTQ<br>man ls<br>ls --help<br>clear<br>mkdir testfolder<br>touch test.txt<br>nano test.txt<br>cp test.txt testfolder<br>mv test.txt test_new.txt<br>mv test_new.txt testfolder<br>rm -r testfolder<br><br>cd FASTQ<br>gunzip SRR7311317_1.fastq.gz<br>head SRR7311317_1.fastq<br>wc -l SRR7311317_1.fastq<br>grep NNN SRR7311317_1.fastq<br>grep NNN SRR7311317_1.fastq &gt; NNN.txt<br>grep NNN SRR7311317_1.fastq | wc -l<br>grep NNN SRR7311317_1.fastq | wc -l &gt;&gt; NNN.txt<br>gzip SRR7311317_1.fastq<br><br># Image of how paired-end reads are obtained<br># <a href="https&#x3a;&#x2F;&#x2F;i0&#x2e;wp&#x2e;com&#x2F;thesequencingcenter&#x2e;com&#x2F;wp&#x2d;content&#x2F;uploads&#x2F;2019&#x2F;03&#x2F;paired&#x2d;end&#x2d;read&#x2d;1&#x2e;jpg&#x3f;resize&#x3d;892&#x25;2C367&amp;ssl&#x3d;1" rel="noreferrer noopener">https:&#x2F;&#x2F;i0.wp.com&#x2F;thesequencingcenter.com&#x2F;wp-content&#x2F;uploads&#x2F;2019&#x2F;03&#x2F;paired-end-read-1.jpg?resize=892%2C367&amp;ssl=1</a><br><br># Recap on RNA-seq pipeline<br># <a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;master&#x2F;vignettes&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;images&#x2F;RNAseq&#x5f;workflow&#x2e;png" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;master&#x2F;vignettes&#x2F;3_RNAseq_workflow&#x2F;images&#x2F;RNAseq_workflow.png</a><br><br><br>## PART1: Quality control of RNAseq reads<br><br># check fastqc arguments by typing this in terminal&#x2F;git bash:<br>fastqc -h<br>## commonly-used arguments:<br># `-o` to specify output directory<br># `-t` to specify the number of CPUs to run program<br><br><br># Today, we will run FastQC with the help of a script<br>## change to scripts directory and create a new script file named run_fastqc.sh<br>cd ~&#x2F;RNAseq_workshop&#x2F;scripts&#x2F;<br>nano run_fastqc.sh<br><br><br>###### SCRIPT SHOULD LOOK LIKE BELOW ######<br>#!&#x2F;bin&#x2F;bash<br><br>echo &quot;Running FastQC analysis......&quot;<br><br>cd ~&#x2F;RNAseq_workshop&#x2F;<br>mkdir .&#x2F;outputs&#x2F;FastQC<br>fastqc .&#x2F;originals&#x2F;FASTQ&#x2F;*.fastq.gz -o .&#x2F;outputs&#x2F;FastQC&#x2F;<br><br>echo &quot;Completed......&quot;<br>##########################################<br><br># Save the contents of the script file by pressing Ctrl+O [that is not a zero] and then enter<br># Exit nano by pressing Ctrl+X<br><br># To run the script, run this command:<br>bash run_fastqc.sh<br><br>ls -l ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC<br><br>## Open the html reports<br>#mac<br>open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_1_fastqc.html<br>open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_2_fastqc.html<br><br><br># windows<br>start ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_1_fastqc.html<br>start ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_2_fastqc.html<br><br># linux<br>xdg-open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_1_fastqc.html<br>xdg-open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC&#x2F;SRR7311317_2_fastqc.html<br><br>## Example of a bad sequencing experiment<br># <a href="https&#x3a;&#x2F;&#x2F;www&#x2e;bioinformatics&#x2e;babraham&#x2e;ac&#x2e;uk&#x2F;projects&#x2F;fastqc&#x2F;bad&#x5f;sequence&#x5f;fastqc&#x2e;html" rel="noreferrer noopener">https:&#x2F;&#x2F;www.bioinformatics.babraham.ac.uk&#x2F;projects&#x2F;fastqc&#x2F;bad_sequence_fastqc.html</a><br><br><br>## TALK ABOUT USING TEXT EDITORS TO EDIT SCRIPTS!!!!!!!!!!<br># <a href="https&#x3a;&#x2F;&#x2F;www&#x2e;sublimetext&#x2e;com&#x2F;" rel="noreferrer noopener">https:&#x2F;&#x2F;www.sublimetext.com&#x2F;</a><br><br><strong>TIme now: 11:11 am</strong><br><strong>Return 12:21 pm&nbsp;</strong><br><br>#### Segway ######<br># I have written a script to download ALL fastq files from this ENA repository (<a href="https&#x3a;&#x2F;&#x2F;www&#x2e;ebi&#x2e;ac&#x2e;uk&#x2F;ena&#x2F;browser&#x2F;view&#x2F;PRJNA476008&#x29;" rel="noreferrer noopener">https:&#x2F;&#x2F;www.ebi.ac.uk&#x2F;ena&#x2F;browser&#x2F;view&#x2F;PRJNA476008)</a><br># Have a look at the script to get an idea on how to perform download files automatically<br># You can configure the script slightly to suit your needs (e.g., changing the link to TSV file)<br># PLEASE MAKE SURE YOU HAVE ENOUGH SPACE ON YOUR COMPTUER TO DOWNLOAD THE DATA!<br># run the following command to download this script<br>cd ~&#x2F;RNAseq_workshop&#x2F;scripts<br>curl -O <a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;master&#x2F;vignettes&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;scripts&#x2F;run&#x5f;bulkfastqdownload&#x2e;sh" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;master&#x2F;vignettes&#x2F;3_RNAseq_workflow&#x2F;scripts&#x2F;run_bulkfastqdownload.sh</a><br>################<br><br>## PART2: Read trimming<br><br># to see trimmomatic arguments<br>trimmomatic -h<br><br># link to trimmomatic home-page<br># <a href="http&#x3a;&#x2F;&#x2F;www&#x2e;usadellab&#x2e;org&#x2F;cms&#x2F;&#x3f;page&#x3d;trimmomatic" rel="noreferrer noopener">http:&#x2F;&#x2F;www.usadellab.org&#x2F;cms&#x2F;?page=trimmomatic</a><br><br># Download all scripts and files required<br>curl -O <a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;master&#x2F;vignettes&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;scripts&#x2F;run&#x5f;trimmomatic&#x2e;sh" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;master&#x2F;vignettes&#x2F;3_RNAseq_workflow&#x2F;scripts&#x2F;run_trimmomatic.sh</a><br>curl -O <a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;master&#x2F;vignettes&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;scripts&#x2F;Nextera&#x5f;TruSeq&#x2d;PE&#x2e;fa" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;master&#x2F;vignettes&#x2F;3_RNAseq_workflow&#x2F;scripts&#x2F;Nextera_TruSeq-PE.fa</a><br><br>cat run_trimmomatic.sh<br><br># Run script<br>bash run_trimmomatic.sh<br><br># check output directories<br>ls -l ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;trimmed_FASTQ&#x2F;<br><br>#### Exercise #####<br># 1. Duplicate the script run_fastqc.sh and rename it<br># 2. Amend the script to now perform FastQC on the trimmed FASTQ files found at ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;trimmed_FASTQ<br># 3. Run the script<br>##################<br><br>Answer:<br>cp run_fastqc.sh run_fastqc_trim.sh<br>nano run_fastqc_trim.sh&nbsp;<br>#### in nano ####<br>#!&#x2F;bin&#x2F;bash<br><br>echo &quot;Running FastQC analysis......&quot;<br><br>cd ~&#x2F;RNAseq_workshop&#x2F;<br>mkdir .&#x2F;outputs&#x2F;FastQC_trimmed<br>fastqc .&#x2F;outputs&#x2F;trimmedFASTQ&#x2F;*.fastq.gz -o .&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;<br><br>echo &quot;Completed......&quot;<br>##########################################<br><ul class="indent"></ul><br># check contents of the new FastQC directory<br>ls -l ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;<br><br># check quality of trimmed reads<br>#mac<br>open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_1.trimmed_fastqc.html<br>open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_2.trimmed_fastqc.html<br><br># windows<br>start ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_1.trimmed_fastqc.html<br>start ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_2.trimmed_fastqc.html<br><br># linux<br>xdg-open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_1.trimmed_fastqc.html<br>xdg-open ~&#x2F;RNAseq_workshop&#x2F;outputs&#x2F;FastQC_trimmed&#x2F;SRR7311317_2.trimmed_fastqc.html<br><br><br>## PART3: Read Mapping using Kallisto<br><br>## Check out arguments to kallisto<br>kallisto<br>kallisto quant --help<br><br># Download all scripts and files required<br>curl -O <a href="https&#x3a;&#x2F;&#x2F;raw&#x2e;githubusercontent&#x2e;com&#x2F;fursham&#x2d;h&#x2F;R&#x2d;datascience&#x2F;master&#x2F;vignettes&#x2F;3&#x5f;RNAseq&#x5f;workflow&#x2F;scripts&#x2F;run&#x5f;kallisto&#x2e;sh" rel="noreferrer noopener">https:&#x2F;&#x2F;raw.githubusercontent.com&#x2F;fursham-h&#x2F;R-datascience&#x2F;master&#x2F;vignettes&#x2F;3_RNAseq_workflow&#x2F;scripts&#x2F;run_kallisto.sh</a><br><br><br>## End of lesson &amp; Feedback<br><br># What is the one thing you learn from this workshop series?<br>Kallisto quantification!<br>Data analysis and trimming<br># What can this workshop improve on in the future?<br>this is as good as it gets<br><br># Would you like a final workshop on Single-cell RNAseq analysis before the summer break?<br>yes, please :)<br>yes, please!<br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br>
</body>
</html>
