---
title: "Setup Instructions"
author: "Fursham Hamid"
output: html_document
---
# Setup required for workshop Day 1 {#bashday1}

## Bash Shell
Bash is a commonly-used shell that gives you the power to do simple tasks more quickly.  
Below are the instructions on installing Shell for the 3 main Computer Operating Systems:

#### Windows
[Video Tutorial](https://www.youtube.com/watch?v=339AEqk9c-8)

1. Download the [Git for Windows installer](https://git-scm.com/downloads).
2. Run the installer and follow the steps below:
    1. Click on "Next" four times (two times if you've previously installed Git). You don't need to change anything in the Information, location, components, and start menu screens.
    2. From the dropdown menu select "Use the nano editor by default" and click on "Next".
    3. Click "Next" on the page titled "Adjusting the name of the initial branch...."
    4. Ensure that "Git from the command line and also from 3rd-party software" is selected and click on "Next". (If you don't do this Git Bash will not work properly, requiring you to remove the Git Bash installation, re-run the installer and to select the "Git from the command line and also from 3rd-party software" option.)
    5. Ensure that "Use the native Windows Secure Channel library" is selected and click on "Next".
    6. Ensure that "Checkout Windows-style, commit Unix-style line endings" is selected and click on "Next".
    7. Ensure that "Use Windows' default console window" is selected and click on "Next".
    8. Click "Next" for the next 2 menu page.
    9. Ensure that "Enable file system caching" and "Enable Git Credential Manager" are selected and click on "Next".
    10. Click "Next" on the menu page titled "Configuring experimental options".
    11. Click on "Install".
    12. Click on "Finish".
3. If your "HOME" environment variable is not set (or you don't know what this is):
    1. Open command prompt (Open Start Menu then type cmd and press [Enter])
    2. Type the following line into the command prompt window exactly as shown:
        `setx HOME "%USERPROFILE%"`
    3. Press [Enter], you should see SUCCESS: Specified value was saved.
    4. Quit command prompt by typing exit then pressing [Enter]

To open the Bash Shell, type "Git Bash" on the search toolbar at the bottom 
of your screen and press enter. This should open a window as shown below.
<p align="center">
  <img src="man/figures/1620376325317.jpg" width="450"/>
</p>

#### MacOS
The default shell in some versions of macOS is Bash, and Bash is available in 
all versions, so no need to install anything. You access Bash from the Terminal 
(found in `/Applications/Utilities`). To open Terminal, press Command-Spacebar 
to open Spotlight search and type "Terminal". You may want to keep Terminal in your 
dock for this workshop.

To see if your default shell is Bash type `echo $SHELL` in Terminal and press 
the enter/return key. If the message printed does not end with '/bash' then 
your default is something else and you can run Bash by typing `bash`.


#### Linux
The default shell is usually Bash and there is usually no need to install anything.

To see if your default shell is Bash type `echo $SHELL` in Terminal and press 
the enter/return key. If the message printed does not end with '/bash' then 
your default is something else and you can run Bash by typing `bash`.

# Setup required for workshop Day 2 {#bashday2}

## Software to process RNAseq data

|Software|Version|Manual|Available for|Description|
|:---|:------------|:------------|:------------|:------------|
|[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)|0.11.9|[Link](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)|Linux, MacOS, Windows|Quality control tool for high throughput sequence data.|
|[Trimmomatic](http://www.usadellab.org/cms/?page=trimmomatic)|0.39|[Link](http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf)|Linux, MacOS, Windows|A flexible read trimming tool for Illumina NGS data.|
|[Kallisto](https://pachterlab.github.io/kallisto/)|0.46.1|[Link](https://pachterlab.github.io/kallisto/manual)|Linux, MacOS, Windows|Quantifying abundances of transcripts from bulk and single-cell RNA-Seq data.|

## Installation

These are the QuickStart installation instructions. They assume familiarity with the command line and with installation in general. As there are different operating systems and many different versions of operating systems and environments, these may not work on your computer. If an installation doesnâ€™t work for you, please refer to the user guide for the tool, listed in the table above.

The easiest way to download these software is using miniconda. Miniconda is a package manager that simplifies the installation process. Please first install miniconda3 (installation instructions below), and then proceed to the installation of individual tools.

### Miniconda3
Miniconda3 is available on Windows, MacOS and Linux operating systems.  
Please download miniconda3 from the this [website](https://docs.conda.io/en/latest/miniconda.html).

<br/>

### FastQC
The easiest way to install FastQC is using miniconda. To install, open your 
Terminal or Command Prompt and type the folling in the console:

```
conda install -c bioconda fastqc=0.11.9
```

You may test your installion by running:
```
fastqc -h
```

<br/>

### Trimmomatic
Below are instructions on installing trimmomatic using miniconda. To install, open your 
Terminal or Command Prompt and type the folling in the console:

```
conda install -c bioconda trimmomatic=0.39=0
```

You may test your installion by running:
```
trimmomatic
```

<br/>

### Kallisto
Below are instructions on installing Kallisto using miniconda. To install, open your 
Terminal or Command Prompt and type the folling in the console:

```
conda install -c bioconda kallisto=0.46.1=0
```

You may test your installion by running:
```
kallisto
```

<br/>
<br/>











