---
title: 'Workshop 3: Intro to RNAseq analysis'
author: "Fursham Hamid"
output: html_document
---

> ### Overview
><br />
> **Duration**: 4 hours (over 2 days)
><br />
><br />
>
> **Objectives**
>
> 1) Describe key reasons for learning shell.
> 2) Navigate your file system using the command line.
> 3) View, search within, copy, move, and rename files. Create new directories.
> 4) Retrieve RNAseq files using shell
> 5) Employ the `grep` command to search for information within files.
> 6) Print the results of a command to a file.
> 7) Explain how a FASTQ file encodes per-base quality scores.
> 8) Interpret a FastQC plot summarizing per-base quality across all reads.
> 9) Clean FASTQ reads using Trimmomatic.
> 10) Align and quantify reads using Kallisto
><br />

## Before starting the lesson
In this workshop, we will be using UNIX Shell to navigate around our computer
filesystem and to process publicly-available RNAseq data. Please refer to this
[setup instructions](https://fursham-h.github.io/R-datascience/articles/3_RNAseq_workflow/installations.html#bashday1) to check if your operating system comes pre-installed
with Shell.  

We will also be using several command-line programs to process reads from RNAseq
experiments. **These programs will be required only for day 2 of the workshop, and
we will briefly go through the installation process during day 1**. For instructions
on installing these software, please refer to this [setup instructions](https://fursham-h.github.io/R-datascience/articles/3_RNAseq_workflow/installations.html#bashday2). 

During the lesson, we will be using etherpad, an online collaborative platform
for our discussions and activities. Below are the links to the lesson pads:

1. [Lesson 1](https://etherpad.wikimedia.org/p/CDN-RNAseqIntro-online)
2. [Lesson 2](https://etherpad.wikimedia.org/p/CDN-RNAseqIntro2-online)

You may bookmark this page before the lesson commence!

<br/ >

## Lesson schedule
**Day 1 [11th May 2021]**

||||
|:---|:------------|:------------|
|10:00|[1. Introducing the Shell](https://datacarpentry.org/shell-genomics/01-introduction/index.html)| What is a command shell and why would I use one? <br>How can I move around on my computer? <br> How can I see what files and directories I have? <br> How can I specify the location of a file or directory on my computer? |
|10:20|[2. Navigating files and directories ](https://datacarpentry.org/shell-genomics/02-the-filesystem/index.html) | How can I perform operations on files outside of my working directory?  <br> What are some navigational shortcuts I can use to make my work more efficient?  |
|10:40|[3. Working with files and directories](https://datacarpentry.org/shell-genomics/03-working-with-files/index.html) | How can I view and search file contents? <br> How can I create, copy and delete files and directories? |
|10:50|**Break**||
|11:00|4. [Retrieving and inspecting RNAseq files](https://datacarpentry.org/wrangling-genomics/02-quality-control/index.html)|  How do I download files from the internet? <br> How do I compress/decompress large files? <br> How do I preview large files in Shell?  |
|11:30|5. [Overview of RNAseq pipeline](https://raw.githubusercontent.com/fursham-h/R-datascience/master/vignettes/3_RNAseq_workflow/images/RNAseq_workflow.png) and installing softwares|  What is the general outline of an RNAseq data analysis? <br> How do I install command-line software?  |
|12:00|Finish| |

<br/>
<br/> 

**Day 2 [25th May 2021]**  

||||
|:---|:------------|:------------|
|10:00|1. Check-in and brief lesson overview| |
|10:15|2. Recap of commands learnt so far| |
|10:20|[3. Overview of RNAseq pipeline ](https://raw.githubusercontent.com/fursham-h/R-datascience/master/vignettes/3_RNAseq_workflow/images/RNAseq_workflow.png) | What is the general workflow of a typical RNAseq pipeline? <br> What tools are available to process RNAseq reads based on my desired research outcome?  |
|10:30|[4. Quality control of RNAseq reads](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) | How do I check the quality of my sequencing reads to make an informed decision on the degree of trimming needed? <br> How do I write a script to perform repititive task? |
|10:50|**Break**||
|11:00|[5. Read trimming](http://www.usadellab.org/cms/?page=trimmomatic)|  How do I trim my reads to obtain high-quality sequences for subsequent steps of the workflow? <br> How do I determine the adaptor sequence to remove from my reads? |
|11:30|[6. Mapping and quantifying reads to the transcriptome](https://pachterlab.github.io/kallisto/manual)| What are indices? <br> How do I determine the strandness of my reads? <br> How do I align my reads to the transcriptome? |
|11:50|Feedback and questions| |






