<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Lesson 2: Manipulating data frames â€¢ CDN.R.Datascience</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><meta property="og:title" content="Lesson 2: Manipulating data frames">
<meta property="og:description" content="CDN.R.Datascience">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">CDN.R.Datascience</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fas fa-book-open"></span>
     
    Lessons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/Lesson_1/Lesson_1.html">1) R Basics for Data Analysis</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fursham-h/R-datascience">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Lesson2_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Lesson 2: Manipulating data frames</h1>
            
      
      
      <div class="hidden name"><code>Lesson2.Rmd</code></div>

    </div>

    
    
<blockquote>
<h2 id="overview">Overview</h2>
<p><br> Teaching: x mins<br>
Exercises: 0 mins</p>
<p><strong>Objectives</strong>:</p>
<ol style="list-style-type: decimal">
<li>Import data into R</li>
<li>Perform basic data manipulation using Tidyverse package</li>
<li>Write readable R code</li>
</ol>
<p><br></p>
</blockquote>
<div id="what-is-a-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#what-is-a-data-frame" class="anchor"></a>What is a data frame?</h2>
<p>A data frame is the most common way of of storing data in R. Technically, it comprises of a list of equal-length vectors. Each element of the list can be thought of as a column and the length of each element of the list is the number of rows.</p>
<p>Most biological data are structured in the same manner as a data frame, i.e. a table that contain one or more column variables and an infinite number of rows listing the values of each variable. A data table can be saved as one of these file types:</p>
<ul>
<li>generic text file: .txt</li>
<li>tab-separated values file: .tsv</li>
<li>comma-separated values file: .csv</li>
<li>excel file: .xls or .xlsx.</li>
</ul>
</div>
<div id="importing-data-into-r-as-data-frame" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-data-into-r-as-data-frame" class="anchor"></a>Importing data into R as data frame</h2>
<p>Converting tabular data into an R data frame is a natural process. In this section, we will learn how to import data tables from your local computer into R.</p>
<p>Create a new RMarkdown file and save it on your local computer. Give the document an appropriate <strong>title</strong> in the metadata section and replace the default content of the document with a header <code>## Import data</code> as in Figure 1. <br></p>
<p><br><br></p>
<p>We begin with retrieving a sample data from the cloud. This file contain a list of alternatively-spliced exons and its inclusion levels in two biological samples; Sfpq mutants and siblings. The data is saved in a tab-delimited file format (tsv). To download this file, create a new R chunk (click <img src="Chunk_insert.PNG">) and type the following code into it:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html">download.file</a></span><span class="op">(</span><span class="st">"https://bit.ly/36J7lx4"</span>, <span class="st">"splicing_data.tsv"</span><span class="op">)</span></code></pre></div>
<p><br></p>
<p>The function <code>download.file</code> will retrieve the content from the input hyperlink and save it to your current working directory under the filename <code>splicing_data.tsv</code>. To import this tsv into R, we will run the <code>read.delim</code> function and provide the name of the file as its input. Add the following code in the chunk above:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span><span class="op">(</span><span class="st">"splicing_data.tsv"</span>, stringsAsFactors <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
<p><br></p>
<p>and run the chunk by pressing the <strong>Run</strong> button (<img src="green-arrow.PNG">). Our data is successfully imported into R and assigned to the variable name <code>dat</code>.</p>
<blockquote>
<h2 id="tip">Tip</h2>
<p><br> In this example, the data file is found in the current working directory, hence providing the name of the file as input to <code>read.delim</code> is sufficient. However, to import files not found in the working directory, you would need to provide the full path to the file as input. To get this information, refer to <a href="https://www.howtogeek.com/670447/how-to-copy-the-full-path-of-a-file-on-windows-10/">this link for Windows user</a> or <a href="https://macpaw.com/how-to/get-file-path-mac">this link for Mac user</a>.</p>
<p><br></p>
</blockquote>
</div>
<div id="previewing-data-frames" class="section level2">
<h2 class="hasAnchor">
<a href="#previewing-data-frames" class="anchor"></a>Previewing data frames</h2>
<p>Once imported, R data frames can be previewed in many different ways.</p>
<p>The first 6 lines or 6 six rows of the data frame <code>dat</code> can be printed using the <code>head</code> or <code>tail</code> function. Create a sub-section in your RMarkdown document and insert these code into a new chunk:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<p>Upon running this chunk, you should get two tables that look like this:</p>
<table class="table">
<thead><tr class="header">
<th align="left">Gene_id</th>
<th align="left">Gene_name</th>
<th align="left">Coord</th>
<th align="left">Strand</th>
<th align="left">Type</th>
<th align="right">Psi_control</th>
<th align="right">Psi_mutant</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45937851-45938028</td>
<td align="left">-</td>
<td align="left">CE</td>
<td align="right">0.89684</td>
<td align="right">0.85626</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45936257-45936592</td>
<td align="left">-</td>
<td align="left">AL</td>
<td align="right">0.12117</td>
<td align="right">0.13731</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45932974-45933558</td>
<td align="left">-</td>
<td align="left">CE</td>
<td align="right">0.91097</td>
<td align="right">0.84387</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45930101-45930150</td>
<td align="left">-</td>
<td align="left">CE</td>
<td align="right">0.87502</td>
<td align="right">0.83428</td>
</tr>
<tr class="odd">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45928441-45928610</td>
<td align="left">-</td>
<td align="left">CE</td>
<td align="right">0.90099</td>
<td align="right">0.87379</td>
</tr>
<tr class="even">
<td align="left">ENSDARG00000036650</td>
<td align="left">cdadc1</td>
<td align="left">1:45928170-45928359</td>
<td align="left">-</td>
<td align="left">CE</td>
<td align="right">0.90005</td>
<td align="right">0.84569</td>
</tr>
</tbody>
</table>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Gene_id</th>
<th align="left">Gene_name</th>
<th align="left">Coord</th>
<th align="left">Strand</th>
<th align="left">Type</th>
<th align="right">Psi_control</th>
<th align="right">Psi_mutant</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">159972</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44642239-44642299</td>
<td align="left">+</td>
<td align="left">CE</td>
<td align="right">0.94589</td>
<td align="right">0.88444</td>
</tr>
<tr class="even">
<td align="left">159973</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44667281-44667315</td>
<td align="left">+</td>
<td align="left">CE</td>
<td align="right">0.93042</td>
<td align="right">0.89334</td>
</tr>
<tr class="odd">
<td align="left">159974</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44667526-44667592</td>
<td align="left">+</td>
<td align="left">CE</td>
<td align="right">0.94962</td>
<td align="right">0.92238</td>
</tr>
<tr class="even">
<td align="left">159975</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44681722-44681925</td>
<td align="left">+</td>
<td align="left">CE</td>
<td align="right">0.37911</td>
<td align="right">0.48088</td>
</tr>
<tr class="odd">
<td align="left">159976</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44683845-44684345</td>
<td align="left">+</td>
<td align="left">AL</td>
<td align="right">0.48199</td>
<td align="right">0.31010</td>
</tr>
<tr class="even">
<td align="left">159977</td>
<td align="left">ENSDARG00000038941</td>
<td align="left">ppp1r1c</td>
<td align="left">9:44686736-44686894</td>
<td align="left">+</td>
<td align="left">CE</td>
<td align="right">0.92443</td>
<td align="right">0.89328</td>
</tr>
</tbody>
</table>
<p>The data frame contain 7 columns, and 159977 rows. The column names are shown at the very top of the table and each row represent one alternative exon. The description of the columns are as such:</p>
<table class="table">
<colgroup>
<col width="18%">
<col width="81%">
</colgroup>
<thead><tr class="header">
<th align="left">Column name</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Gene_id</td>
<td align="left">Given ID of the parental gene</td>
</tr>
<tr class="even">
<td align="left">Gene_name</td>
<td align="left">Gene name of the parental gene</td>
</tr>
<tr class="odd">
<td align="left">Coord</td>
<td align="left">Genomic position of exon</td>
</tr>
<tr class="even">
<td align="left">Strand</td>
<td align="left">Whether exon is in the <code><a href="https://rdrr.io/r/base/Arithmetic.html">+</a></code> or <code><a href="https://rdrr.io/r/base/Arithmetic.html">-</a></code> strand</td>
</tr>
<tr class="odd">
<td align="left">Type</td>
<td align="left">Type of alternative exon<br>CE: Casette exon<br>AF: Alternative first<br>AL:Alternative last<br>AD: Alternative donor<br>AA: Alternative acceptor<br>RI: Retained intron</td>
</tr>
<tr class="even">
<td align="left">Psi_control</td>
<td align="left">Percent spliced in index of exon in control</td>
</tr>
<tr class="odd">
<td align="left">Psi_mutant</td>
<td align="left">Percent spliced in index of exon in mutant</td>
</tr>
</tbody>
</table>
<p>The first 6 rows of the data frame can also be shown in a compact way using <code>str</code> function. Type the following code in a new code chunk and run it:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<p>The above code should output:</p>
<pre><code>##                Gene_id            Gene_name                      Coord       
##  ENSDARG00000099378:   260   zgc:174563:   260   1:10003490-10003530:     1  
##  ENSDARG00000099959:   168   smyhc1    :   168   1:10004609-10004693:     1  
##  ENSDARG00000067997:   150   myhz1.3   :   150   1:10007900-10008011:     1  
##  ENSDARG00000069692:   148   col7a1l   :   148   1:10036183-10036319:     1  
##  ENSDARG00000032630:   131   neb       :   131   1:10040088-10040216:     1  
##  ENSDARG00000093780:   126   (Other)   :153535   1:10043648-10043803:     1  
##  (Other)           :158994   NA's      :  5585   (Other)            :159971  
##  Strand    Type         Psi_control          Psi_mutant       
##  -:81807   AA:  5041   Min.   :0.0000881   Min.   :0.0001125  
##  +:78170   AD:  4230   1st Qu.:0.8958800   1st Qu.:0.8755300  
##            AF:  1920   Median :0.9593600   Median :0.9482000  
##            AL:   816   Mean   :0.8776923   Mean   :0.8705174  
##            CE:145191   3rd Qu.:0.9846200   3rd Qu.:0.9809100  
##            RI:  2779   Max.   :0.9999800   Max.   :0.9999700  
## </code></pre>
<p>To summarize the contents of each column, we can run the <code>summary</code> function. Create a new code chunk and run the following code:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">dat</span><span class="op">)</span></code></pre></div>
<p>which should look like this:</p>
<pre><code>##                Gene_id            Gene_name                      Coord       
##  ENSDARG00000099378:   260   zgc:174563:   260   1:10003490-10003530:     1  
##  ENSDARG00000099959:   168   smyhc1    :   168   1:10004609-10004693:     1  
##  ENSDARG00000067997:   150   myhz1.3   :   150   1:10007900-10008011:     1  
##  ENSDARG00000069692:   148   col7a1l   :   148   1:10036183-10036319:     1  
##  ENSDARG00000032630:   131   neb       :   131   1:10040088-10040216:     1  
##  ENSDARG00000093780:   126   (Other)   :153535   1:10043648-10043803:     1  
##  (Other)           :158994   NA's      :  5585   (Other)            :159971  
##  Strand    Type         Psi_control          Psi_mutant       
##  -:81807   AA:  5041   Min.   :0.0000881   Min.   :0.0001125  
##  +:78170   AD:  4230   1st Qu.:0.8958800   1st Qu.:0.8755300  
##            AF:  1920   Median :0.9593600   Median :0.9482000  
##            AL:   816   Mean   :0.8776923   Mean   :0.8705174  
##            CE:145191   3rd Qu.:0.9846200   3rd Qu.:0.9809100  
##            RI:  2779   Max.   :0.9999800   Max.   :0.9999700  
## </code></pre>
</div>
<div id="basic-manipulation-1-filter" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-manipulation-1-filter" class="anchor"></a>Basic manipulation 1: Filter</h2>
</div>
<div id="basic-manipulation-2-select" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-manipulation-2-select" class="anchor"></a>Basic manipulation 2: Select</h2>
</div>
<div id="basic-manipulation-3-mutate" class="section level2">
<h2 class="hasAnchor">
<a href="#basic-manipulation-3-mutate" class="anchor"></a>Basic manipulation 3: Mutate</h2>
</div>
<div id="the-pipe" class="section level2">
<h2 class="hasAnchor">
<a href="#the-pipe" class="anchor"></a>The pipe <code>%&gt;%</code>
</h2>
<blockquote>
<h2 id="practice-on-your-own">Practice on your own</h2>
<p><br> The Markdown syntax contain various formattings that enhances the body of your document. Read this <a href="https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet">CheatSheet</a> and attempt the following tasks in a new RMarkdown document.</p>
<ol style="list-style-type: decimal">
<li>Create an <strong>ordered list</strong> containing 3 of your favorite food</li>
<li>Bold the #1 fav from the above list</li>
<li>Insert an image of your #1 fav food. Can be from a local image on your computer or from a hyperlink.</li>
</ol>
<p>Render the created RMarkdown as HTML and compare your output with mine (<a href="https://bit.ly/3cFtYX3">found here</a>).</p>
<p><br></p>
</blockquote>
<p><br><br><br><br><br><br></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Fursham Hamid.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
